nrf-hal-common/src/uarte.rs

```diff
@@ -406,18 +440,250 @@ pub enum Error {
     BufferNotInRAM,
 }
 
-pub trait Instance: Deref<Target = uarte0::RegisterBlock> + sealed::Sealed {}
+pub trait Instance: Deref<Target = uarte0::RegisterBlock> + sealed::Sealed {
+    fn ptr() -> *const uarte0::RegisterBlock;
```

Comment:
Is that it already? Shall I just fork @lulf then and "re PR" this I guess?

Reply:
I should take a closer look at this, but that's one thing I noticed

Reply:
So after reading a bit through the current implementation I saw that it still uses this Deref super trait https://github.com/nrf-rs/nrf-hal/blob/master/nrf-hal-common/src/uarte.rs#L452 (as well as other peripherals) so I'm quite unsure how to get rid off it? (I do get why the ptr() can be get rid off of course). I'm unfortunately not really familiar with the project so could you go into a bit of detail as to why we don't need it here?

Reply:
I'd like to remove the `Deref` supertrait, since it's essentially redundant with the `ptr` method. I guess this doesn't have to be done in *this* PR though.